Core/include/Acts/TrackFitting/Chi2Fitter.hpp

```diff
@@ -0,0 +1,734 @@
+        nUpdates(nIter),
+        calcFinalChi2(calcFinalChi2_),
+        logger(logger_) {}
+  /// Contexts are required and the options must not be default-constructible.
+  Chi2FitterOptions() = delete;
+
+  /// Context object for the geometry
+  std::reference_wrapper<const GeometryContext> geoContext;
+  /// Context object for the magnetic field
+  std::reference_wrapper<const MagneticFieldContext> magFieldContext;
+  /// context object for the calibration
+  std::reference_wrapper<const CalibrationContext> calibrationContext;
+
+  Chi2FitterExtensions extensions;
+
+  /// The trivial propagator options
+  PropagatorPlainOptions propagatorPlainOptions;
+
+  /// Whether to consider multiple scattering
+  bool multipleScattering = false;  // TODO: add later
+
+  /// Whether to consider energy loss
+  bool energyLoss = false;  // TODO: add later
+
+  /// Number of iterations to improve chi2
+  int nUpdates;
+
+  /// Whether to do an additional propagation step, just to get the latest chi2
+  /// value
+  bool calcFinalChi2;
+
+  /// Logger
+  LoggerWrapper logger;
+
+  /// debug only: event number of the current event, passed by
+  /// TrackFittingChi2Algorithm
+  // TODO: remove me
+  size_t debugEventNumber = 0;
+};
+
+struct Chi2FitterResult {
+  // Fitted states that the actor has handled.
+  MultiTrajectory fittedStates;
+
+  // This is the index of the 'tip' of the track stored in multitrajectory.
+  // This correspond to the last measurment state in the multitrajectory.
+  // Since this KF only stores one trajectory, it is unambiguous.
+  // SIZE_MAX is the start of a trajectory.
+  size_t lastMeasurementIndex = SIZE_MAX;
+
+  // This is the index of the 'tip' of the states stored in multitrajectory.
+  // This correspond to the last state in the multitrajectory.
+  // Since this KF only stores one trajectory, it is
+  // unambiguous. SIZE_MAX is the start of a trajectory.
+  size_t lastTrackIndex = SIZE_MAX;
+
+  // The optional Parameters at the provided surface
+  std::optional<BoundTrackParameters> fittedParameters;
+  // TODO: use Acts::Chi2FitterExtensions::Parameters?
+
+  // Counter for states with non-outlier measurements
+  size_t measurementStates = 0;  // TODO: use to handle finalization
+
+  // Counter for measurements holes
+  // A hole correspond to a surface with an associated detector element with no
+  // associated measurment. Holes are only taken into account if they are
+  // between the first and last measurements.
+  size_t measurementHoles = 0;
+
+  // Counter for handled states
+  size_t processedStates = 0;
+
+  // Indicator if track fitting has been done
+  bool finished = false;
+  // we do not have smoothed, reset, reversed, like the KF does.
+
+  bool isFinalized = false;
+
+  // Measurement surfaces without hits
+  std::vector<const Surface*> missedActiveSurfaces;
+
+  // collectors
+  std::vector<ActsScalar>
+      m_measurements_collector;  // pixel layers add 2 entries, strip layers
```

Comment:
I think these varaibles should be renamed following the naming conventions in Acts.